src/snowflake/cli/_plugins/snowpark/snowpark_entity_model.py

```diff
@@ -59,17 +47,24 @@ class SnowparkEntityModel(EntityModelBase, ExternalAccessBaseModel, ImportsBaseM
         title="Python version to use when executing ", default=None
     )
     stage: str = Field(title="Stage in which artifacts will be stored")
-    artifacts: List[Union[PathMapping, str]] = Field(title="List of required sources")
+    artifacts: Artifacts = Field(title="List of required sources")
 
     @field_validator("artifacts")
     @classmethod
     def _convert_artifacts(cls, artifacts: Union[dict, str]):
         _artifacts = []
-        for artefact in artifacts:
-            if isinstance(artefact, PathMapping):
-                _artifacts.append(artefact)
+        for artifact in artifacts:
+            if (
+                "*" in artifact
+                and FeatureFlag.ENABLE_SNOWPARK_BUNDLE_MAP_BUILD.is_disabled()
```

Comment:
nit: not generally a fan of naming publicly visible concepts by implementation details like "bundle map". Is there a more user-friendly name we could use here?

Also, is it not the default because of backwards compat concerns? I wonder if the solution isn't to make it default, but then give a flag to revert to old behaviour if needed (SNOWPARK_LEGACY_GLOB_BEHAVIOUR=true ?)

Reply:
It can not be default, because it is BCR. New build changes zip paths.

Reply:
Maybe `SNOWPARK_GLOB_SUPPORT`?